### ISSUE TITLE
(React) Animation work

### ISSUE THREAD

**Issue by Awattling** [Awattling/React-Login-Reusable#4]
Make it pretty

**Comment by petercr** [comment]
What kind of animation work are you looking for exactly? :thinking: 
I could be into writing some CSS for it

**Comment by Awattling** [comment]
Right now all the animations are from bootstrap. Any modifications would be subtle things to make the UI more intuitive. (fades, hover effects, highlights). I'm thinking things like border highlights to indicate fields that are passed or failed as things are typed. On registration a password strength indicator would be nice. A loading Icon and support for things in the process of loading. Perhaps a nice set of animations as the UI is loaded and changed. (Hit make account and the make account box is faded in instead of the hard re-render). There's a lot here and some are trickier than others. You may need to modify states and props of components and perhaps use JavaScript to add/removes classes. Take what you can chew and all improvements are appreciated.

**Comment by petercr** [comment]
Are you open to adding other **JavaScript** libraries for the animations, or did you want to go with straight **CSS**? 
Either way it could be a fun little project. :sunglasses: :smile: 
Maybe you could start a project board here on GitHub, and that would help to organize the tasks :+1:

**Comment by Awattling** [comment]
I'm open to adding other libraries so long as the application remains pretty lightweight. JQuery is a huge library which I would like to avoid. Things like React's CSSTransitionGroup etc are great.

**Comment by petercr** [comment]
I totally understand wanting to step away from **_jQuery._**  :wink: :+1: 
There is another really cool project called [Reactstrap  <= Visit Site ](https://reactstrap.github.io/)
Which is a **React** project where they use Bootstrap components **_but without jQuery_**
You should check it out, it's a pretty cool library I've been meaning to get into.
Maybe we could switch the project over from Bootstrap to Reactstrap ?? :thinking:

**Comment by Awattling** [comment]
Yeah I've heard of that library before and done a little bit of work with it. I'm totally down for transitioning this project over to Reactstrap on the predicate that it continues to work well with CSS Modules. I find CSS modules to be a way better way of handling CSS on a large scale and would not want to sacrifice that functionality.

**Comment by petercr** [comment]
Yes either I figured maybe we could swap out one component at a time if we(you) brought the package in.
With the animations I've been looking into some pretty cool libraries for React Animations.
Like [Pop Motion Pose](https://popmotion.io/pose/learn/popmotion-get-started/)
You can check it out if you want. 
I'll let you know how it's going :+1:  I've got the project up and running, so that's a start

**Comment by Awattling** [comment]
Was it hard to get up and running? I should create a "get up and running" guide for those just entering react. Sadly I have to take a break from this project as something hit my desk that might get me paid but I should be back working on this within the week. Your more than welcome to get started and I'll merge in when I can. Animation wise I'm up for anything that can easily be understood and adopted by those entering react. It looks like Pop Motion is pretty simple and can act as a great starting point.

**Comment by petercr** [comment]
So I **AM** having a little trouble getting the CSS modules and Pop Motion Pose to work together. 
Still trying to figure out how exactly to be able to mix the 2 together, but I'm still working on it :smile: 
_Also I've been sick :face_with_thermometer:  for a week so that has not helped_
